log4j-api/src/main/java/org/apache/logging/log4j/spi/CopyOnWriteSortedArrayThreadContextMap.java

```diff
@@ -267,12 +267,9 @@ public boolean equals(final Object obj) {
         final Map<String, String> map = this.getImmutableMapOrNull();
         final Map<String, String> otherMap = other.getImmutableMapOrNull();
         if (map == null) {
-            if (otherMap != null) {
-                return false;
-            }
-        } else if (!map.equals(otherMap)) {
-            return false;
+            return otherMap == null;
+        } else {
+            return map.equals(otherMap);
```

Comment:
This can even be simplified further: `return map == null ? otherMap == null : map.equals(otherMap)`.